lib/Frontend/CompilerInvocation.cpp

```diff
@@ -930,6 +899,41 @@ static bool ParseSearchPathArgs(SearchPathOptions &Opts,
   return false;
 }
 
+static void ValidateLocatizationArgs(ArgList &Args, DiagnosticEngine &Diags) {
+  using namespace options;
+
+  // Check if the locale code is available.
+  if (Arg *A = Args.getLastArg(OPT_locale)) {
+    std::string localeCode = A->getValue();
+
+    if (!llvm::any_of(localeCodes, [&](const char *locale) {
+          return localeCode == locale;
+        })) {
+      std::string availableLocaleCodes = "";
+      llvm::interleave(
+          std::begin(localeCodes), std::end(localeCodes),
+          [&](std::string locale) { availableLocaleCodes += locale; },
+          [&] { availableLocaleCodes += ", "; });
+
+      // Incorrect locale code shouldn't crash the compiler,
+      // and the compiler will fallback to the default messages.
+      Diags.diagnose(SourceLoc(), diag::error_invalid_locale_code,
+                     availableLocaleCodes);
+    } else {
+      // Check if the localization path exists but it doesn't have a file
+      // for the specified locale code.
+      if (Arg *LA = Args.getLastArg(OPT_localization_path)) {
+        llvm::SmallString<128> localizationPath(LA->getValue());
+        llvm::sys::path::append(localizationPath, localeCode);
+        llvm::sys::path::replace_extension(localizationPath, ".yaml");
+        if (!llvm::MemoryBuffer::getFileOrSTDIN(localizationPath))
+          Diags.diagnose(SourceLoc(), diag::cannot_find_locale_file, localeCode,
```

Comment:
If the locale code is (an ISO 639-1 code) but we still don't support localization for this language, yet. We will show the warning that we cannot find a localization file for the specified locale code. Does this make sense?

Reply:
Sure! If there is no translations then we should warn to make sure that it's recorded somewhere why localization didn't work.